kolibri/core/assets/src/views/BottomNavigationBar.vue

```diff
@@ -0,0 +1,124 @@
+<template>
+
+  <div class="bottom-bar" :style="{ backgroundColor: $themeTokens.textInverted }">
+    <span v-for="(routeDefinition, key) in bottomMenuOptions[0].routes" :key="key">
+      <a
+        :href="generateNavRoute(routeDefinition.route, bottomMenuOptions[0].url)"
+        tabindex="-1"
+        class="nav-menu-item"
+        :style="{ textDecoration: 'none' }"
+        @click="handleNav(routeDefinition.route)"
+      >
+        <div
+          :style="isActiveLink(routeDefinition.route, bottomMenuOptions[0].url) ?
+            bottomMenuActiveStyles :
+            bottomMenuInactiveStyles"
+        >
+          <KIconButton
+            :icon="routeDefinition.icon"
+            :color="$themeTokens.primary"
+            :ariaLabel="routeDefinition.label"
+            size="small"
+          />
+        </div>
+        <p
+          v-if="isActiveLink(routeDefinition.route, bottomMenuOptions[0].url)"
+          class="nav-menu-label"
+          :style="{ color: $themeTokens.primary }"
+        >
+          {{ routeDefinition.label }}
+        </p>
+      </a>
+    </span>
+    <span class="nav-menu-item" :style="bottomMenuInactiveStyles">
+      <KIconButton
+        icon="menu"
+        :ariaLabel="coreString('menuLabel')"
+        :color="$themeTokens.primary"
+        @click="$emit('toggleNav')"
+      />
+      <p :style="{ color: $themeTokens.primary }">{{ coreString('menuLabel') }}</p>
+    </span>
+  </div>
+
+</template>
+
+
+<script>
+
+  import commonCoreStrings from '../mixins/commonCoreStrings';
+  import { generateNavRoute } from '../utils/generateNavRoutes';
+
+  export default {
+    name: 'BottomNavigationBar',
+    mixins: [commonCoreStrings],
+    props: {
+      bottomMenuOptions: {
+        type: Array,
+        required: true,
+      },
+    },
+    computed: {
+      bottomMenuActiveStyles() {
+        return {
+          borderTop: `4px solid ${this.$themeTokens.primary}`,
+        };
+      },
+      bottomMenuInactiveStyles() {
+        return {
+          borderTop: `4px solid ${this.$themeTokens.textInverted}`,
+        };
+      },
+    },
+    methods: {
+      generateNavRoute(route, url) {
+        const params = this.$route.params;
+        return generateNavRoute(url, route, params);
+      },
+      isActiveLink(route, url) {
+        return window.location.pathname === url && route == this.$router.currentRoute.path;
```

Comment:
I think route might be a vue-router route specification, in which case it will include parameter names that will never be in the path.

Reply:
As we are in the SPA in question here, we can just check the route.name against `this.$router.name`, right?